### PR TITLE
REP-2607 Add X-Impersonator-Roles Header

### PR DESCRIPTION
**Pull request by dmnjohns**

### CODE REVIEW

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java [hunk 1]
```diff
@@ -165,6 +165,15 @@ private void setImpersonator() {
...
             filterDirector.requestHeaderManager().putHeader(OpenStackServiceHeader.IMPERSONATOR_ID.toString(), cachableToken.getImpersonatorTenantId());
         }
+
+        if (!cachableToken.getImpersonatorRoles().isEmpty()) {
+            StringBuilder roles = new StringBuilder();
+            for (String role : cachableToken.getImpersonatorRoles()) {
+                roles.append(role);
+                roles.append(',');
+            }
+            filterDirector.requestHeaderManager().appendHeader(OpenStackServiceHeader.IMPERSONATOR_ROLES.toString(), roles.substring(0, roles.length() - 1));
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java]: le sigh, java makes me a sad panda

File: repose-aggregator/components/filters/derp/src/main/scala/org/openrepose/filters/derp/DerpFilter.scala [hunk 2]
```diff
@@ -36,7 +36,8 @@ import scala.util.{Failure, Success}
  * This filter is header quality aware; the delegation header with the highest quality will be used to formulate a
  * response.
  */
-class DerpFilter extends Filter with HttpDelegationManager with LazyLogging {
+class
+DerpFilter extends Filter with HttpDelegationManager with LazyLogging {
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/derp/src/main/scala/org/openrepose/filters/derp/DerpFilter.scala]: what?

File: repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java (hunk 1 above)
Comment (dmnjohns) [inline repose-aggregator/components/filters/client-auth/src/main/java/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManager.java]: I thought the exact same thing when I was writing it.

File: repose-aggregator/components/filters/derp/src/main/scala/org/openrepose/filters/derp/DerpFilter.scala (hunk 2 above)
Comment (dmnjohns) [inline repose-aggregator/components/filters/derp/src/main/scala/org/openrepose/filters/derp/DerpFilter.scala]: Literally no idea how this got in here. Will remove.

File: repose-aggregator/components/filters/client-auth/src/test/groovy/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManagerGroovyTest.groovy [hunk 3]
```diff
@@ -50,6 +50,7 @@ class OpenStackAuthenticationHeaderManagerGroovyTest extends Specification {
         wwwAuthHeaderContents = "test URI";
         endpointsBase64 = "endpointsBase64";
         authToken = Mock()
+        authToken.getImpersonatorRoles() >> new HashSet<String>()
```
Comment (adrianjgeorge) [inline repose-aggregator/components/filters/client-auth/src/test/groovy/org/openrepose/filters/clientauth/openstack/OpenStackAuthenticationHeaderManagerGroovyTest.groovy]: spock mock makes me sad

**Pull request merged**